thoth/storages/graph/postgres.py

```diff
@@ -1194,6 +1197,119 @@ def create_advised_software_stack_pipfile(
+                        build_hardware_information_id=build_hardware_information.id,
+                        run_software_environment_id=run_software_environment.id,
+                        run_hardware_information_id=run_hardware_information.id,
+                    )
+                    self._session.upsert(inspection_run)
+
+                else:
+                    inspection_run, _ = InspectionRun.get_or_create(
+                        self._session,
+                        inspection_sync_state="SYNCED",
+                        inspection_document_id=inspection_document_id,
+                        datetime=document.get("created"),
+                        amun_version=None,  # TODO: propagate Amun version here which should match API version
+                        build_requests_cpu=build_cpu,
+                        build_requests_memory=build_memory,
+                        run_requests_cpu=run_cpu,
+                        run_requests_memory=run_memory,
+                        build_software_environment_id=build_software_environment.id,
+                        build_hardware_information_id=build_hardware_information.id,
+                        run_software_environment_id=run_software_environment.id,
+                        run_hardware_information_id=run_hardware_information.id,
+                    )
+
+                if document["specification"].get("script"):  # We have run an inspection job.
+
+                    if not document["job_log"]["stdout"]:
+                        raise ValueError("No values provided for inspection output %r", inspection_document_id)
+
+                    performance_indicator_name = document["job_log"]["stdout"].get("name")
+                    performance_model_class = PERFORMANCE_MODEL_BY_NAME.get(performance_indicator_name)
+
+                    if not performance_model_class:
+                        raise PerformanceIndicatorNotRegistered(
+                            f"No performance indicator registered for name {performance_indicator_name!r}"
+                        )
+                    framework = document["job_log"]["stdout"].get("framework")
+                    if not framework:
+                        _LOGGER.warning(
+                            "No machine learning framework specified in performance indicator %r", cls.__name__
+                        )
+
+                    overall_score = document["job_log"]["stdout"].get("overall_score")
+                    if overall_score is None:
+                        _LOGGER.warning("No overall score detected in performance indicator %r", overall_score)
+
+                    performance_indicator = performance_model_class.create_from_report(self._session, document)
+
+                    if performance_indicator_name == "PiMatmul":
+                        pi_matmul, _ = PiMatmul.get_or_create(
+                            performance_indicator, inspection_run_id=inspection_run.id
+                        )
```

Comment:
Is this if statement required? It looks to me the above call `performance_indicator_name` should handle creation for all the indicators.